src/app/container/version-modal/version-modal.component.ts

```diff
@@ -0,0 +1,242 @@
+  styleUrls: ['./version-modal.component.css']
+})
+export class VersionModalComponent implements OnInit, AfterViewChecked {
+  public TagEditorMode = TagEditorMode;
+  public DescriptorType = DescriptorType;
+  public isModalShown: boolean;
+  private editMode;
+  private mode: TagEditorMode;
+  private tool: any;
+  private unsavedVersion;
+  private savedCWLTestParameterFiles: Array<any>;
+  private savedWDLTestParameterFiles: Array<any>;
+  private savedCWLTestParameterFilePaths: Array<string>;
+  private savedWDLTestParameterFilePaths: Array<string>;
+  private unsavedCWLTestParameterFilePaths: Array<string>;
+  private unsavedWDLTestParameterFilePaths: Array<string>;
+  private unsavedTestCWLFile = '';
+  private unsavedTestWDLFile = '';
+  private formErrors = formErrors;
+  private version: ToolVersion;
+
+  tagEditorForm: NgForm;
+  @ViewChild('tagEditorForm') currentForm: NgForm;
+
+  constructor(
+    private paramfilesService: ParamfilesService,
+    private versionModalService: VersionModalService,
+    private listContainersService: ListContainersService,
+    private containerService: ContainerService,
+    private containerTagsService: ContainerTagsWebService,
+    private stateService: StateService,
+    private dateService: DateService) {
+  }
+
+  // Almost all these functions should be moved to a service
+  getSizeString(size) {
+    return this.versionModalService.getSizeString(size);
+  }
+
+  onSubmit() {
+    switch (this.mode) {
+      case TagEditorMode.Add: {
+        this.addTag();
+        break;
+      }
+      case TagEditorMode.Edit: {
+        this.editTag();
+        break;
+      }
+      default: {
+        console.log('No idea how you submitted in neither edit or add mode');
+      }
+    }
+  }
+  addTag() {
+    console.log('Saving tag...');
+  }
+  ngAfterViewChecked() {
+    this.formChanged();
+  }
+
+  formChanged() {
+    if (this.currentForm === this.tagEditorForm) { return; }
+    this.tagEditorForm = this.currentForm;
+    if (this.tagEditorForm) {
+      this.tagEditorForm.valueChanges
+        .subscribe(data => this.onValueChanged(data));
+    }
+  }
+
+  onValueChanged(data?: any) {
+    if (!this.tagEditorForm) { return; }
+    const form = this.tagEditorForm.form;
+    for (const field in formErrors) {
+      if (formErrors.hasOwnProperty(field)) {
+        // clear previous error message (if any)
+        formErrors[field] = '';
+        const control = form.get(field);
+        if (control && !control.valid) {
+          const messages = validationMessages[field];
+          for (const key in control.errors) {
+            if (control.errors.hasOwnProperty(key)) {
+              formErrors[field] += messages[key] + ' ';
+            }
+          }
+        }
+      }
+    }
+  }
+
+  editTag() {
+    console.log('Editing tag...');
+    const newCWL = this.unsavedCWLTestParameterFilePaths.filter(x => this.savedCWLTestParameterFilePaths.indexOf(x) === -1);
```

Comment:
Might be a clearer way of doing this like includes https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Array/includes